client/options.go

```diff
@@ -194,10 +197,10 @@ func (o *options) GetInstrumentOptions() instrument.Options {
 	return o.instrumentOpts
 }
 
-func (o *options) EncodingTsz() Options {
+func (o *options) EncodingM3TSZ() Options {
 	opts := *o
 	opts.readerIteratorAllocate = func(r io.Reader) encoding.ReaderIterator {
-		return tsz.NewReaderIterator(r, tsz.NewOptions())
+		return m3tsz.NewReaderIterator(r, defaultIntOptimizationEnabled, encoding.NewOptions())
```

Comment:
`defaultIntOptimizationEnabled` is defined both here and in `storage/options.go`. Can we make it part of the encoding options so you can use it via something like `encoding.NewOptions().GetIntOptimizationEnabled()` down below?